docs/src/charm/tutorial/getting-started.md

```diff
@@ -1 +1,222 @@
+Before we start doing things in Kubernetes, we should consider adding a worker.
+The K8s worker is an additional node for the cluster which focuses on running workloads without running any
+control-plane services. This means it needs a connection to a control-plane node to tell
+it what to do, but it also means more of its resources are available for
+running workloads. We can deploy a worker node in a similar way to the original
+K8s node:
+
+```
+juju deploy k8s-worker --channel=latest/edge --constraints='cores=2 mem=16G root-disk=40G'
+```
+
+Once again, this will take a few minutes. In this case though, the `k8s-worker`
+application won't settle into a 'Ready' status, because it requires a
+connection to the control plane. This is handled in Juju by integrating the
+charms so they can communicate using a standard interface. The charm info we
+fetched earlier also includes a list of the relations possible, and from this
+we can see that the k8s-worker requires "cluster: k8s-cluster"
+
+To connect these charms and effectively add the worker to our cluster, we use the 'integrate' command, adding the interface
+we wish to connect
+
+```
+juju integrate k8s k8s-worker:cluster
+```
+
+After a short time, the worker node will share information with the control plane and be joined to the cluster.
+
+## 4. Scale the cluster (Optional)
+If one worker doesn't seem like enough, we can easily add more:
+
+```
+juju add-unit k8s-worker -n 1
+```
+
+This will create an additional instance running the k8s-worker charm. Juju
+manages all instances of the same application together, so there is no need to
+add the integration again. If you check the Juju status, you should see that a
+new unit is created, and now you have `k8s-worker/0` and `k8s-worker/1`
+
+
+## 5. Use kubectl
+
+Kubectl is the standard upstream tool for interacting with a Kubernetes cluster. This is the command that can be used to inspect and manage your cluster.
+
+If you don't already have `kubectl`, it can be installed from a snap:
+
+```
+sudo snap install kubectl --classic
+```
+
+If you have just installed it, you should also create a file to contain the configuration
+
+```
+mkdir ~/.kube
+```
+
+To fetch the configuration information from the cluster we can run
+
+```
+juju run k8s/0 get-kubeconfig 
+```
+
+The Juju action is a piece of code which runs on a unit to perform a specific
+task. In this case it collects the cluster information - the YAML formatted
+details of the cluster and the keys required to connect to it.
+
+```{warning}  If you already have Kubectl and are using it to manage other clusters,
+   you should edit the relevant parts of the cluster yaml output and add them to
+   your current config.
+```
+
+We can pipe this output directly to a config file which will just require a bit of editing:
+
+```
+juju run k8s/0 get-kubeconfig >> ~/.kube/config
+```
+
+The output includes the root of the YAML, `kubeconfig: |`, so we can just use an editor to remove that line:
```

Comment:
I feel that what you've got here is more clear that having the user pipe the output through jq and writing it to a file. 

but here's the nonsense anyway
```bash
juju run k8s/0 get-kubeconfig | yq '.kubeconfig' -r > .kube/config
```

Reply:
Ah yes, cool, thansk for this. Perhaps we could get the action to do that and then we wouldn't need to do either :joy:

Reply:
UGH. i wish. I don't think a juju action gives us the ability to write a key output to a file like this